Wasm SDK packed as a nuget package

maraf opened the pull request:
Move bits of Wasm SDK that are expected to change more frequently to the runtime repo.

- Add `Microsoft.NET.Sdk.WebAssembly.Pack` implicit package reference if `UsingMicrosoftNETSdkWebAssembly=true`.
- Provide `_WebAssemblyPropsFile` and `_WebAssemblyTargetsFile` properties to enable overriding props and targets.
- Move tests for Wasm SDK tasks to runtime repo. Leave tests checking boot.json in place at the moment.
- Return some default property values back to Blazor SDK to support .NET 5.

Contributes to https://github.com/dotnet/runtime/issues/81367

<!--
TODO
- [x] Blocked on https://github.com/dotnet/msbuild/issues/8478 
- [x] Depends on https://github.com/dotnet/runtime/pull/84082
- [x] Depends on https://github.com/dotnet/runtime/pull/84924 to flow to installer
- [x] Depends on https://github.com/dotnet/installer/pull/16159
- [x] Update SDK
- [x] Blazor tests that are using Wasm tasks
- [x] ~Figure out if we can have an SDK without dll with tasks~ Empty tasks dll is how others do it
- [x] Use `UsingMicrosoftNETSdkWebAssembly` to trigger adding `Microsoft.NET.Sdk.WebAssembly.Pack` reference
-->

maraf committed:
Remove most of Wasm SDK bits

maraf committed:
Fix KnownWebAssemblySdkPacks in FrameworkReferenceResolution

maraf committed:
Import using overridable _WebAssemblyPropsFile and _WebAssemblyTargetsFile.

maraf committed:
Move wasm tasks related tests to runtime. Leave BootJsonData related tests in place at the moment

maraf committed:
Don't override existing RuntimeIdentifier in SDK.props. Pass UsingMicrosoftNETSdkWebAssembly to ProcessFrameworkReferences.

maraf committed:
Move BootJsonData for tests to tests project.

maraf committed:
Rename KnownWebAssemblySdkPack to KnownWebAssemblySdkBrowserPack

maraf committed:
Revert "Rename KnownWebAssemblySdkPack to KnownWebAssemblySdkBrowserPack"

This reverts commit 0af6292e08063a172a6e2b3515e1c8e709c99591.

maraf committed:
Drop check for RuntimeIdenfier.

Include WasmSDK Pack based on UsingMicrosoftNETSdkWebAssembly

maraf committed:
Localization

maraf committed:
Merge remote-tracking branch 'upstream/main' into WasmSdkPackaged

maraf committed:
UpdateXlf

lewing commented on src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs:
```diff
@@ -688,13 +703,13 @@ private enum ToolPackType
 
             // ILCompiler and ILLink have RID-agnostic build packages that contain MSBuild targets.
-            if (toolPackType is ToolPackType.ILCompiler or ToolPackType.ILLink)
+            if (toolPackType is ToolPackType.ILCompiler or ToolPackType.ILLink or ToolPackType.WebAssemblySdk)
```
Not a strong preference here but it seems like
```suggestion
            if (toolPackType is not ToolPackType.Crossgen2)
```
is slightly cleaner

lewing reviewed (commented):
This is looking good, what else is left?

maraf committed:
Update SDK

maraf committed:
Feedback

maraf commented:
> This is looking good, what else is left?

maraf committed:
Remove local nuget source.

maraf committed:
Merge remote-tracking branch 'upstream/main' into WasmSdkPackaged

# Conflicts:
#	src/WasmSdk/Targets/Microsoft.NET.Sdk.WebAssembly.Current.targets

maraf committed:
Fix after merge with main

maraf committed:
Update installer version

maraf committed:
Merge remote-tracking branch 'upstream/main' into WasmSdkPackaged

maraf committed:
Dont fail when WasmPack is not found (= building older .NET project)

maraf committed:
Fix property expression. UsingBlazorSdk is still probably needed.

maraf committed:
[ToRevert] Temporarily hook Compression

maraf committed:
Merge remote-tracking branch 'upstream/main' into WasmSdkPackaged

maraf committed:
Update SDK

maraf committed:
Revert "[ToRevert] Temporarily hook Compression"

This reverts commit fcc5d48d9b9585a87c7eb1c3d2cab44898b5a4c1.

maraf committed:
Force browser-wasm on BlazorWebAssemblySdk.

WasmSdk supports browser-wasm and wasi-wasm, but Blazor typically wants to override RID passed for server project. That is not the case for WasmSdk.

maraf committed:
Force AppendRuntimeIdentifierToOutputPath on BlazorWebAssemblySdk.

It is shared for net50 blazor.
WasmSdk might turn it off to support different RIDs.

maraf committed:
Force CopyOutputSymbolsToPublishDirectory on BlazorWebAssemblySdk.

It is shared for net50 blazor.

maraf committed:
Update comment

lewing commented:
@rainersigwald @tmat @mthalman I'm having a little trouble following all the flow here but are we expecting the failures here to be resolved once everthing flows? https://github.com/dotnet/sourcelink/issues/1007

tmat commented:
@lewing I think so.

lewing committed:
bump the sdk

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/sourcelink build 20230421.2

Microsoft.Build.Tasks.Git , Microsoft.SourceLink.AzureRepos.Git , Microsoft.SourceLink.Bitbucket.Git , Microsoft.SourceLink.Common , Microsoft.SourceLink.GitHub , Microsoft.SourceLink.GitLab
 From Version 8.0.0-beta.23210.3 -> To Version 8.0.0-beta.23221.2

lewing commented:
triggered https://github.com/dotnet/sdk/pull/31953 but also pulled the bump in here to verify

lewing commented:
@javiercn @MackinnonBuck any objections?
